ocrd_network/ocrd_network/processing_worker.py

```diff
@@ -0,0 +1,317 @@
+            result_message = OcrdResultMessage(
+                job_id=str(job_id),
+                status=job_state.value,
+                # Either path_to_mets or workspace_id must be set (mutually exclusive)
+                path_to_mets=processing_message.path_to_mets,
+                workspace_id=None
+            )
+            self.log.debug(f'Result message: {result_message}')
+            encoded_result_message = OcrdResultMessage.encode_yml(result_message)
+            self.rmq_publisher.publish_to_queue(
+                queue_name=processing_message.result_queue,
+                message=encoded_result_message
+            )
+
+    def run_processor_from_worker(
+            self,
+            processor_class,
+            workspace,
+            page_id: str,
+            input_file_grps: List[str],
+            output_file_grps: List[str],
+            parameter: dict,
+    ) -> bool:
+        input_file_grps_str = ','.join(input_file_grps)
+        output_file_grps_str = ','.join(output_file_grps)
+
+        success = True
+        try:
+            run_processor(
+                processorClass=processor_class,
+                workspace=workspace,
+                page_id=page_id,
+                parameter=parameter,
+                input_file_grp=input_file_grps_str,
+                output_file_grp=output_file_grps_str,
+                instance_caching=True
+            )
+        except Exception as e:
+            success = False
+            self.log.exception(e)
+
+        if not success:
+            self.log.error(f'{processor_class} failed with an exception.')
+        else:
+            self.log.debug(f'{processor_class} exited with success.')
+        return success
+
+    def run_cli_from_worker(
+            self,
+            executable: str,
+            workspace,
+            page_id: str,
+            input_file_grps: List[str],
+            output_file_grps: List[str],
+            parameter: dict
+    ) -> bool:
+        input_file_grps_str = ','.join(input_file_grps)
+        output_file_grps_str = ','.join(output_file_grps)
+
+        return_code = run_cli(
+            executable=executable,
+            workspace=workspace,
+            page_id=page_id,
+            input_file_grp=input_file_grps_str,
+            output_file_grp=output_file_grps_str,
+            parameter=json.dumps(parameter),
+            mets_url=workspace.mets_target
+        )
+
+        if return_code != 0:
+            self.log.error(f'{executable} exited with non-zero return value {return_code}.')
+            return False
+        else:
+            self.log.debug(f'{executable} exited with success.')
+            return True
+
+    def set_job_state(self, job_id: Any, state: StateEnum):
+        """Set the job status in mongodb
+        """
+        # TODO: the way to interact with mongodb needs to be thought about. Beanie seems not
```

Comment:
The proposed solution reduces the readability of the code. Instead, there is a `call_sync` [wrapper](https://github.com/OCR-D/ocrd-webapi-implementation/blob/main/ocrd_webapi/utils.py#L175-L184) implemented in the reference WebAPI implementation that was not adopted here yet.

Reply:
Your solution looks nice to me. Did you try if it works with `beanie`? Why isn't it used here yet?

Reply:
> Your solution looks nice to me. Did you try if it works with beanie?

Yes. The workers (in the service broker module) in Operandi use `call_sync` version of db methods coming from the WebAPI reference implementation. 

> Why isn't it used here yet?

Because the `set_job_state` was implemented before introducing the `call_sync` to the WebAPI. Another reason is that currently there is no connection preservation to anything. Connections are created when needed and closed afterward. Yet another reason was to keep this implementation simple since job state update is the only thing happening in this PR.

Reply:
If the db client connection is closed the processing worker will also fail. But the way it is implemeneted currently - a new client connection is created with every job status update - which is of course bad for performance.

Although, I am not sure about the details of the beanie db client and whether it automatically restores the connection.

Reply:
And why does this stop you from implementing something like `call_sync` in `set_job_state`, so that `beanie` can be used instead of `pymongo`?

Reply:
> Yet another reason was to keep this implementation simple since job state update is the only thing happening in this PR.

Do you want this PR merged soon or?

Reply:
Yes, but it doesn't stop me from optimizing the code and understanding the problem. From my understanding, it's not complicated to implement that function like this, or?

```python
@call_sync
async def set_job_state(self, job_id: Any, state: StateEnum):
    job = await Job.get(job_id)
    job.state = state
    await job.save()
```

Reply:
Yes, simple but testing if it breaks something takes time - anyway, I put that time in. The `call_sync` is now available.